openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -371,66 +636,145 @@ def _polymer_openmm_topology_to_rdmol(
                         if sorted_match not in sorted_mol_frags:
                             continue
 
-                    # Some special residues are allowed to overlap/override previous matches
-                    if any(m in already_assigned_nodes for m in match) and (
-                        res_name
-                        not in [
-                            "PEPTIDE_BOND",
-                            "DISULFIDE",
-                            "UNIQUE_MOLECULE",
-                            "ADDITIONAL_SUBSTRUCTURE",
-                        ]
-                    ):
-                        continue
-                    already_assigned_nodes.update(match)
+                    # assign chemical info and check all overlapping substructures for ambiguous/conflicting chemical info
 
-                    for atom_i, j in zip(ref.GetAtoms(), match):
+                    for atom_i, j in zip(ref_info.GetAtoms(), full_match):
+                        if atom_i.GetAtomicNum() == 0:  # ignore neighboring atoms
+                            continue
                         atom_j = mol.GetAtomWithIdx(j)
+                        # error checking for overlapping substructures with priority. Enforce that no ambiguous chemical assignments are made
+                        if (
+                            j in already_assigned_nodes
+                        ):  # if overlapping with previous match
+                            if (
+                                atom_i.GetFormalCharge() != atom_j.GetFormalCharge()
+                                and atom_i.GetIdx() not in sym_atoms
+                            ):
+                                error_reason = f"Formal charge of new query ({atom_i.GetFormalCharge()}) does not match the formal charge of previous query ({atom_j.GetFormalCharge()})"
+                                raise AmbiguousAtomChemicalAssignment(
+                                    res_name,
+                                    atom_j.GetIdx(),
+                                    atom_i.GetIdx(),
+                                    error_reason,
+                                )
+                            elif atom_i.GetChiralTag() != atom_j.GetChiralTag():
+                                error_reason = f"Chiral Tag of new query ({atom_i.GetChiralTag()}) does not match the chiral tag of previous query ({atom_j.GetChiralTag()})"
+                                raise AmbiguousAtomChemicalAssignment(
+                                    res_name,
+                                    atom_j.GetIdx(),
+                                    atom_i.GetIdx(),
+                                    error_reason,
+                                )
                         # copy over chirality
                         if atom_i.GetChiralTag():
                             mol.GetAtomWithIdx(j).SetChiralTag(atom_i.GetChiralTag())
                         atom_j.SetFormalCharge(atom_i.GetFormalCharge())
```

Comment:
I actually had some uncertainty in coding this originally. It was added because the original amino acid templates had stereochemistry. However, if we only get stereo from 3d, I agree that this is pointless. Would you support removing all mention of stereochem before the final assignment from 3d coordinates?

Reply:
Gotcha - if you're also doubtful about this then I'll remove it. I've confirmed that we're still doing AssignStereochemistryFrom3D afterwards anyway, so this shouldn't cause a conflict.